test/formal/checked_int_add.py

```diff
@@ -0,0 +1,39 @@
+from rule import Rule
+from opcodes import *
+from util import *
+
+"""
+Overflow checked signed integer addition.
+"""
+
+n_bits = 256
+type_bits = 8
+
+while type_bits <= n_bits:
+
+	rule = Rule()
+
+	# Input vars
+	X_short = BitVec('X', type_bits)
+	Y_short = BitVec('Y', type_bits)
+
+	# Z3's overflow and underflow conditions
+	actual_overflow = (BVAddNoOverflow(X_short, Y_short, True) == False)
```

Comment:
I assume the `==` returns `z3::Bool`, just wondering if we should be explicit about `False`

Reply:
Could also write
`Not(BVAddNoOverflow(X_short, Y_short, True))`

Reply:
Yeah - here we could change it to ``Not`` - I don't mind. The ``== False`` might be slightly easier to read due to the double negation in "Not(No-Overflow)", but it's not a huge difference.

Reply:
Changed it to ``Not`` in any case.